internal/ip/database/database.go

```diff
@@ -0,0 +1,35 @@
+// package database provides tools for enriching IP addresses from enrichment databases.
+package database
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/brexhq/substation/internal/errors"
+	"github.com/brexhq/substation/internal/ip"
+)
+
+// errInvalidFactoryInput is returned when an unsupported Databaser is referenced in Factory.
+const errInvalidFactoryInput = errors.Error("invalid factory input")
+
+// Databaser provides tools for opening, managing, and reading enrichment information from IP address enrichment databases.
+type Databaser interface {
```

Comment:
style: The interface name of `database.Databaser` is sort of redundant and doesn't specify what what it does, following [Effective Go](https://go.dev/doc/effective_go#interface-names) and the [io](https://go.dev/doc/effective_go#interface-names) package, what about naming this `database.ReadOpenCloser`?

Reply:
agreed the stutter is bad, fixed by breaking up interface